packages/@react-spectrum/label/stories/HelpText.stories.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Flex} from '@react-spectrum/layout';
+import React, {useState} from 'react';
+import {SpectrumTextFieldProps} from '@react-types/textfield';
+import {storiesOf} from '@storybook/react';
+import {TextField} from '@react-spectrum/textfield';
+
+storiesOf('HelpText', module)
+  .addParameters({providerSwitcher: {status: 'positive'}})
+  .add(
+    'description',
+    () => render({description: 'Password must be at least 8 characters.'})
+  )
+  .add(
+    'error message',
+    () => render({errorMessage: 'Create a password with at least 8 characters.', validationState: 'invalid'})
+  )
+  .add(
+    'description and error message',
+    () => {
+      let [value, setValue] = useState('');
+
+      return (
+        <TextField
+          label="Empty field"
+          description="This input is only valid when it's empty."
+          errorMessage="Remove input."
+          value={value}
+          onChange={setValue}
+          validationState={value.length ? 'invalid' : undefined} />
+      );
+    }
+  )
+  .add(
+    'description, validationState: valid',
+    () => render({
+      label: 'Nickname',
+      description: 'Enter your nickname, or leave blank if you don\'t have one.',
+      validationState: 'valid'
+    })
+  )
+  .add(
+    'description and error message, validationState: valid',
+    () => render({
+      label: 'Valid field',
+      description: 'The error message will never render because validationState is "valid".',
+      errorMessage: 'Uninformative error message', // Won't render
+      validationState: 'valid'
+    })
+  )
+  .add(
+    'showErrorIcon: true',
```

Comment:
Thanks, good catch. I think I misinterpreted the the issue requirement that "All of the components that support help text will extend from this interface" — I had them extend from `SpectrumHelpTextProps` instead of `HelpTextProps`. Looks like `SpectrumHelpTextProps` was meant just for the HelpText component like you mentioned in your other comment